python/tank/platform/software_launcher.py

```diff
@@ -336,47 +375,83 @@ def get_standard_plugin_environment(self):
 
         return env
 
-    def is_version_supported(self, version):
+    def is_version_supported(self, sw_version):
```

Comment:
In terms of the naming, I was thinking of the `version` in `is_version_supported` as a `SoftwareVersion`. But if we change the name of that object, it would make sense to do that here. Perhaps the best solution is to add simple convenience methods to check `_is_version_supported(version)` and `_is_product_supported(product)` which check the corresponding property list. And we also add `_is_software_version_supported(sw_version)` which uses these convenience methods. It feels like in reality though, if calling code wants to check if a version or product, is supported, they'd just check against the properties. What do you guys think?